src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -510,7 +510,14 @@ void ScenesServer::GroupWillBeRemoved(FabricIndex aFabricIx, EndpointId aEndpoin
     SceneTable * sceneTable = scenes::GetSceneTableImpl(aEndpointId);
     VerifyOrReturn(nullptr != sceneTable);
 
-    MakeSceneInvalid(aEndpointId);
+    chip::GroupId currentGroup;
+    Attributes::CurrentGroup::Get(aEndpointId, &currentGroup);
+
+    // Validate if the Current Group = RemovedGroup before invalidating the scene.
```

Comment:
```suggestion
    // If currentGroup is what is being removed, we can't possibly still have a valid scene,
    // because the scene we have (if any) will also be removed.
```